oauth/middleware.py

```diff
@@ -0,0 +1,26 @@
+from django.contrib.auth import authenticate
+from django.http import HttpResponse
+
+
+class OAuth2TokenMiddleware:
+    """
+    Django 1.10 has changed the middlewares signature.
+    oauth2_provider.middleware.OAuth2TokenMiddleware is still old style.
+    https://github.com/evonove/django-oauth-toolkit/issues/406
```

Comment:
Humm, this still looks like a hack to me anyway. It seems weird to me to be using a third party oauth lib, and needing to:
- check for "Bearer" by hand (not sure this work for all oauth flows)
- set request.user by hand
- return a 401 when not auth
- deal with WWW-Authenticate header

Imho, either this should be committed upstream, either we should change third party app, either we should review our needs.
What do you think?